cPP_DSC.adoc

```diff
@@ -2536,82 +2490,99 @@ This component is targeted at the flaw remediation procedures applied by the dev
 
-FCS_CKM.1.1/AKG:: The TSF shall generate *asymmetric* cryptographic keys in accordance with a specified cryptographic key generation algorithm [*selection*: _cryptographic key generation algorithm_] and specified cryptographic key sizes [*selection*: _cryptographic key sizes_] that meet the following: [*selection*: _list of standards_].
+FCS_CKM.1.1/AKG:: The TSF shall generate *asymmetric* cryptographic keys in accordance with a specified cryptographic key generation algorithm [*selection*: _cryptographic key generation algorithm_] and specified cryptographic *algorithm parameters* [.line-through]#key sizes# {empty}[*selection*: _cryptographic algorithm parameters_] that meet the following: [*selection*: _list of standards_].
 
-.Allowed choices for completion of the selection operations of FCS_CKM.1/AKG.
+.Asymmetric Cryptographic Key Generation
 [[AsymKeyGen]]
-[cols=".^1,.^2,.^2,.^2",options=header]
+[cols=".^1,.^2,.^2",options=header]
 |===
 
-|Identifier
-|Key Type
-|Key Sizes
+|Cryptographic Key Generation Algorithm
+|Cryptographic Algorithm Parameters
 |List of Standards
 
-|AK1 
 |RSA 
-|{empty}[*selection*: [.underline]#2048 bit, 3072 bit]#
-|FIPS PUB 186-5 (Section A.1.1)
+|Modulus of size [selection2048 bit, 3072 bit]
+|NIST FIPS PUB 186-5 (Section A.1.1)
 
-|AK2 
 |ECDSA - Extra Random Bits 
-|{empty}[*selection*: [.underline]#256 bits (P-256, brainpoolP256r1), 384 bits (P-384, brainpoolP384r1), 512 bits (P-521, brainpoolP512r1)]#
-|[*selection*: FIPS PUB 186-5 (Section A.2.1), NIST SP 800-56A (Section 5.6.1.2.1)]
+|Elliptic curve [selection: 256 bits (P-256, brainpoolP256r1), 384 bits (P-384, brainpoolP384r1), 512 bits (P-521, brainpoolP512r1)]
+|[selection: NIST FIPS PUB 186-5 (Section A.2.1), NIST SP 800-56A r3 (Section 5.6.1.2.1)]
+
+[selection: NIST SP 800-186 (Section 4) [NIST Curves], RFC 5639 (section 3) [brainpool curves]]
 
-|AK3 
 |ECDSA - Rejection Sampling
-|{empty}[*selection*: [.underline]#256 bits (P-256, brainpoolP256r1), 384 bits (P-384, brainpoolP384r1), 512 bits (P-521, brainpoolP512r1)]#
-|[*selection*: FIPS PUB 186-5 (Section A.2.2), NIST SP 800-56A (Section 5.6.1.2.2)]
+|Elliptic curve [selection: 256 bits (P-256, brainpoolP256r1), 384 bits (P-384, brainpoolP384r1), 512 bits (P-521, brainpoolP512r1)]
+|[selection: NIST FIPS PUB 186-5 (Section A.2.2), NIST SP 800-56A r3 (Section 5.6.1.2.2)]
 
-|AK4 
-|DSA 
-|Bit lengths of p and q respectively (L, N) {empty}[*selection*: [.underline]#(3072, 256)]#
-|BSI TR-02102-1 (Section 6.3.2) [Key generation]
+[selection: NIST SP 800-186 (Section 4) [NIST Curves], RFC 5639 (section 3) [brainpool curves]]
+
+|FFC - Extra Random Bits
+|Static domain parameters approved for [selection: IKE groups [selection: MODP-2048, MODP-3072, MODP-4096, MODP-6144, MODP-8192], TLS groups [selection: ffdhe2048, ffdhe3072, ffdhe4096, ffdhe6144, ffdhe8192]]]
+|NIST SP 800-56Ar3 [approved FFC domain parameters] 
+
+NIST SP 800-56A (Section 5.6.1.1.3) [key pair generation]
```

Comment:
Not sure why 56A is referenced twice here. I would also reference RFC 3526 and 7919 here for the groups.

Reply:
@slpotte

Reply:
There are two steps here, the first is choosing the domain parameters, the second is in generating the private and public keys.

The crypto working group tended to list IETF documents as a last resort when no other documents existed.  I think including here will be appropriate since the actual domain parameters are listed in the RFC.  I've added both RFCs to both rows in the catalog.

Reply:
@slpotte can you paste the text in here so I can update the current iteration of the doc to match?

Reply:
@slpotte I see this is RFC 7919, but are you saying that you list it as a selection (like in the EdDSA link) or as something more like replacing the "approved FFC domain parameters" with something like "parameters as specified in RFC 7919"?

Reply:
NIST SP 800-56A Rev. 3, RFC 3526, RFC 7919 [approved FFC domain parameters]

Reply:
Maybe we should remove the word "approved" and leave it as "[FFC domain parameters]"?